### ISSUE TITLE
Material Dobouts

### ISSUE THREAD

**Issue by RodrigoMedeirosrRS** [tobspr/RenderPipeline#66]
Hello!

I am using the Render Pipeline with a egg file to load a 3D animated character, i easily can load the model and set the basic parameters, and here is result:
![antes](https://cloud.githubusercontent.com/assets/7956669/21485248/51d5f868-cb87-11e6-9e9f-5e785dd2d035.png)
But i want use normal maps, and different types of shading models (in this case Default and Skin), i use the material editor to configure the scene, and here is the result:
![depois](https://cloud.githubusercontent.com/assets/7956669/21485312/639a7d70-cb88-11e6-8903-f17a16feb50a.png)
But the modifications of Material Editor are temporary, so i want know how i can modify this specific materials into script or file, to make this modifications permanently?

**Comment by RodrigoMedeirosrRS** [comment]
Right i solve it!
Into the Egg file, the parameters:
emitR = Shader Model
emitG = Normal Strenght
emitB = ?

Now i have one important dobout, how i cam use the Emissive maps? My game is futuristic, and this  map is realy important for me.

**Comment by tobspr** [comment]
Hi, 
The reason the material data is encoded in the emissive component is that panda doesnt currently support storing additional Information on materials. 

If you install my bam exporter, it offers the ability to modify those parameters in blender (even if you export it later with yabee). 

Right now the rp does not support emissive maps. A material can have only one shading model at a time. You can use two materials though, one for the emissive parts, and one for the regular parts. 

If you want to control the strength of the emission on materials with the emissive shading model, you can simply modify the basecolor (the first texture slot). 

Let me know if this works for you. 

FYI, in future (experimental)  versions,  theres also a material api class to create and modify those materials with a nicer interface. 

Best regards, 
Tobias

**Comment by RodrigoMedeirosrRS** [comment]
Right, i got it. I am already using the bam exporter and the egg exporter.

But if you accept a suggestion, please add a Alpha and a texture multiplier to Emissive Shadding Mode.

Solid color is more useful for artificial lights, but if i want create glowing fire or thunder bilboard, or a magma material, the solid emission is useless. But if i can only change the solid color of Emissive with a texture and apply a alpha on this material, his will become much more useful.

**Comment by tobspr** [comment]
I am currently overhauling transparency, and in future Versions (experimental)  there will be a seperate emissive transparency model, since for particles you might want to have both as you mentioned. 

Reason that I did not merge it yet is that I also made some architectural changes, so not all plugins work in the experimental branch.

**Comment by RodrigoMedeirosrRS** [comment]
Very Cool! :)
